src/main/scala/com/fulcrumgenomics/sopt/cmdline/CommandLineProgramParser.scala

```diff
@@ -227,19 +227,34 @@ class CommandLineProgramParser[T](val targetClass: Class[T], val includeSpecialA
       .filter { withSpecial || !_.isSpecial }
       .partition { !_.optional }
 
-    if (required.nonEmpty) {
-      builder.append(KRED(s"\n${KBLDRED(targetName)} ${KRED(RequiredArguments)}\n"))
-      builder.append(lineBreak(color=true))
-      new ClpArgumentLookup(required:_*).ordered.foreach { arg =>
-        ClpArgumentDefinitionPrinting.printArgumentDefinitionUsage(builder, arg, argumentLookup)
+    // Split the arguments by group
+    val groups = argumentLookup.view
+      .filterNot { _.hidden }
+      .filter { withSpecial || !_.isSpecial }
+      .groupBy(_.groupName.getOrElse(""))
+
+    // Print out the commands by group
+    groups.keys.toSeq.sorted.map(_.capitalize).foreach { groupName =>
+      val group: Seq[ClpArgument] = groups(groupName)
+      val (required, optional) = group.partition { !_.optional }
```

Comment:
I also am beginning to wonder if the required/optional split is a good idea if there are multiple groups.  It leads to a lot of sections.  I'm not sure what else to suggest though that would make it clear what's optional and what's required.